src/azure-cli/azure/cli/command_modules/botservice/commands.py

```diff
@@ -80,5 +86,11 @@ def load_command_table(self, _):
     with self.command_group('bot webchat', channelOperations_commandType) as g:
         g.command('show', 'webchat_get')
 
+    with self.command_group('bot directline site', directLineSitesOperations_commandType, is_preview=True) as g:
+        g.command('create', 'create')
+        g.command('delete', 'delete')
+        g.command('list', 'list')
+        g.command('update', 'update')
```

Comment:
generic_update_command ?

Reply:
I didn’t the generic update command because the site is a complex sub-resource with no direct API.

The update API called here is for all Channels (Direct Line, MS Teams, Slack, etc). The Direct Line channel’s properties contains a list of sites, so to preserve any pre-existing sites the module needs to fetch the channel and store them for the lifetime of the command.